e2e/test/iothub/properties/PropertiesE2ETests.cs

```diff
@@ -0,0 +1,448 @@
+                .ConfigureAwait(false);
+        }
+
+        [LoggedTestMethod]
+        public async Task Properties_DeviceSendsNullValueForPropertyResultsServiceRemovingIt_MqttWs()
+        {
+            await Properties_DeviceSendsNullValueForPropertyResultsServiceRemovingItAsync(
+                    Client.TransportType.Mqtt_WebSocket_Only)
+                .ConfigureAwait(false);
+        }
+
+        [LoggedTestMethod]
+        public async Task Properties_ClientHandlesRejectionInvalidPropertyName_Mqtt()
+        {
+            await Properties_ClientHandlesRejectionInvalidPropertyNameAsync(
+                    Client.TransportType.Mqtt_Tcp_Only)
+                .ConfigureAwait(false);
+        }
+
+        [LoggedTestMethod]
+        public async Task Properties_ClientHandlesRejectionInvalidPropertyName_MqttWs()
+        {
+            await Properties_ClientHandlesRejectionInvalidPropertyNameAsync(
+                    Client.TransportType.Mqtt_WebSocket_Only)
+                .ConfigureAwait(false);
+        }
+
+        private async Task Properties_DeviceSetsPropertyAndGetsItBackSingleDeviceAsync(Client.TransportType transport)
+        {
+            TestDevice testDevice = await TestDevice.GetTestDeviceAsync(Logger, _devicePrefix).ConfigureAwait(false);
+            using var deviceClient = DeviceClient.CreateFromConnectionString(testDevice.ConnectionString, transport);
+
+            await Properties_DeviceSetsPropertyAndGetsItBackAsync(deviceClient, testDevice.Id, Guid.NewGuid().ToString(), Logger).ConfigureAwait(false);
+        }
+
+        private async Task Properties_DeviceSetsPropertyMapAndGetsItBackSingleDeviceAsync(Client.TransportType transport)
+        {
+            TestDevice testDevice = await TestDevice.GetTestDeviceAsync(Logger, _devicePrefix).ConfigureAwait(false);
+            using var deviceClient = DeviceClient.CreateFromConnectionString(testDevice.ConnectionString, transport);
+
+            await Properties_DeviceSetsPropertyAndGetsItBackAsync(deviceClient, testDevice.Id, s_mapOfPropertyValues, Logger).ConfigureAwait(false);
+        }
+
+        public static async Task Properties_DeviceSetsPropertyAndGetsItBackAsync<T>(DeviceClient deviceClient, string deviceId, T propValue, MsTestLogger logger)
+        {
+            string propName = Guid.NewGuid().ToString();
+
+            logger.Trace($"{nameof(Properties_DeviceSetsPropertyAndGetsItBackAsync)}: name={propName}, value={propValue}");
+
+            var props = new ClientPropertyCollection();
+            props.AddRootProperty(propName, propValue);
+            await deviceClient.UpdateClientPropertiesAsync(props).ConfigureAwait(false);
+
+            // Validate the updated properties from the device-client
+            ClientProperties clientProperties = await deviceClient.GetClientPropertiesAsync().ConfigureAwait(false);
+            bool isPropertyPresent = clientProperties.TryGetValue<T>(propName, out T propFromCollection);
```

Comment:
propValueFromCollection?

Reply:
sure, I don't mind renaming this variable